src/non-matching-media-query-remover.js

```diff
@@ -69,7 +69,11 @@ function nonMatchingMediaQueryRemover (
       if (csstree.keyword(atrule.name).name !== 'media') {
         return
       }
-
+      // this can happen - why? (atrule.prelude === null)
+      // and should we remove this rule here, or keep it?
+      if (!atrule.prelude) {
```

Comment:
@lahmatiy This seems to come from #222 - came across this error in production, have not yet mapped it to original css, but the prelude was `null`, and `children` was `[]`.. anything you can say already just based on that info? (should this check be needed etc)

Reply:
This may be the case if there is an error in CSS, e.g. `@media { }` – no prelude here and children is empty. Since CSSTree parser is tolerant to errors it doesn't validate a CSS for correctness of specific at-rules or properties, it just make a check of basic structures (i.e. at-rule can contain a prelude and a block, but both are optional). Later you can use lexer to validate AST against the CSS rules. At the moment lexer can validate a declarations, support for other parts of CSS (like at-rules) will be added in next releases and it will allow to filter a bad nodes before the rest processing.
You can set [`positions: true`](https://github.com/csstree/csstree/blob/master/docs/parsing.md#positions) option to `parse()` and look at `loc` field of a specific node to find a source of problem.